_posts/2025-04-28-sparse-autodiff.md

```diff
@@ -749,6 +749,32 @@ This backup storage enables the use of **fewer distinct colors**, reducing the c
 Powell and Toint <d-cite key="powellEstimationSparseHessian1979"></d-cite> were the first to notice symmetry-related optimizations, before Coleman and Moré <d-cite key="colemanEstimationSparseHessian1984"></d-cite> made the connection to graph coloring explicit.
 While symmetric coloring and decompression are more computationally expensive than their nonsymmetric counterparts, this cost is typically negligible compared to the savings we get from fewer HVPs.
 
+## Applications
+
+ASD is useful in applications which require the computation of full Jacobian or Hessian matrices with sparsity.
+One such example is given in the 2024 ICLR blog post *How to compute Hessian-vector products?* <d-cite key="dagreouHowComputeHessianvector2024"></d-cite>,
+where Hessians are used for second-order optimization:
+
+> When trying to find the minimum of the function $$f$$, methods that account for the second-order information often rely on the product between the inverse Hessian and a vector to find a good update direction.
+> For instance, Newton's method relies on update rules of the form
+> 
+> \begin{equation}
+>   \notag
+>   \theta_{k+1} = \theta_k - \eta_k[\nabla^2f(\theta_k)]^{-1}\nabla f(\theta_k)
+> \end{equation}
+> 
+> for some step-size $$\eta_k>0$$.
+
+The blog post goes on to argue that instead of materializing the full Hessian matrix, 
+matrix-free iterative solvers should be combined with Hessian operators by iteratively computing HVPs: 
+
+> When evaluating the term $$[\nabla^2f(\theta_k)]^{-1}\nabla f(\theta_k)$$, it would be very inefficient to first compute the full Hessian matrix $$\nabla^2f(\theta_k)$$, then invert it and finally multiply this with the gradient $$\nabla f(\theta_k)$$.
+
+While this holds true for dense Hessians, the opposite can be the case for sparse Hessians.
+Materializing a full Hessian matrix allows for an entirely new class of linear solvers to be used, e.g. enabling matrix factorization.
+Whether these are more performant depends on the number of colors in the sparity pattern of the Hessian 
+vs. the required numerical precision of the iterative solve.
```

Comment:
Iterative solvers are typically a last resort when memory limitations prevent the use of ASD.

Reply:
So you're saying we can make even stronger claims here?

Reply:
Could you elaborate on this in the blog post?

Reply:
I suggest https://github.com/adrhill/iclr2025-sparse-autodiff/pull/60#discussion_r1976999620